clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -462,17 +462,10 @@ private boolean maybeSendAndPollTransactionalRequest() {
             return true;
         }
 
-        if (transactionManager.hasAbortableError() || transactionManager.isAborting()) {
-            if (accumulator.hasIncomplete()) {
```

Comment:
Do we still want to include the accumulator.hasIncomplete check around the two other statements?

Reply:
Right. It's a no-op so check is not necessary.